File: Python/8kyu/8kyu-count-the-monkeys.py
# Count the Monkeys!
# Language: Python
# Difficulty: 8 kyu

def monkey_count(n):
    return list(range(1, n + 1))


# Tests
import codewars_test as test
from solution import monkey_count

@test.describe("Fixed Tests")
def basic_tests():
    test.describe("Basic tests")
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(monkey_count(5), [1, 2, 3, 4, 5])
        test.assert_equals(monkey_count(3), [1, 2, 3])
        test.assert_equals(monkey_count(9), [1, 2, 3, 4, 5, 6, 7, 8, 9])
        test.assert_equals(monkey_count(10), [1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
        test.assert_equals(monkey_count(20), [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20])

File: R/8kyu/8kyu-count-the-monkeys.r
# Count the Monkeys!
# Language: R
# Difficulty: 8 kyu

# Solution 1
monkey_count <- function(n) {
  seq(1, n)
}

# Solution 2
monkey_count <- function(n) {
  1:n
}


# Tests
test_that('basic tests', {
  expect_equal(monkey_count(5), c(1, 2, 3, 4, 5))
  expect_equal(monkey_count(3), c(1, 2, 3))
  expect_equal(monkey_count(9), c(1, 2, 3, 4, 5, 6, 7, 8, 9))
  expect_equal(monkey_count(10), c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
  expect_equal(monkey_count(20), c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20))
})